### PR TITLE
chore(e2e): replace direct 'curl' usage with util

### PR DESCRIPTION
**Pull request by lobkovilya**
This PR replaces all `curl` usages in e2e tests the except usages like:
```go
cmd := []string{"/bin/bash", "-c", "\"curl localhost:9901/clusters | grep test-server-mtls\""}
```

### Checklist prior to review

- [X] [Link to relevant issue][1] as well as docs and UI issues --
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [X] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: test/e2e/cni/taint_controller_race.go [hunk 1]
```diff
@@ -83,27 +84,29 @@ metadata:
...
-				_, stderr, err := cluster.Exec(TestNamespace, clientPodName, "demo-client",
-					"curl", "-v", "-m", "3", "--fail", "test-server.kuma-test.svc.80.mesh")
-				return stderr, err
-			}, "10s", "1s").Should(ContainSubstring("HTTP/1.1 200 OK"))
+			Eventually(func(g Gomega) {
+				_, err := client.CollectResponse(
+					cluster, "demo-client", "test-server",
+					client.FromKubernetesPod(TestNamespace, "demo-client"),
+				)
+				g.Expect(err).ToNot(HaveOccurred())
```
Comment (bartsmykla) [inline test/e2e/cni/taint_controller_race.go:92]: Maybe we could go with something like:
```suggestion
			Eventually(func(g Gomega) {
				g.Expect(client.CollectResponse(
					cluster, "demo-client", "test-server",
					client.FromKubernetesPod(TestNamespace, "demo-client"),
				)).Error().ToNot(HaveOccurred())
```
Wdyt?

**Review by bartsmykla — APPROVED** [review]

File: test/e2e_env/kubernetes/k8s_api_bypass/k8s_api_bypass.go [hunk 2]
```diff
@@ -72,9 +83,14 @@ spec:
...
-			stdout, _, err := kubernetes.Cluster.Exec(namespace, clientPodName, "demo-client", "bash", "-c", cmd)
+			stdout, _, err := client.CollectRawResponse(
+				kubernetes.Cluster, "demo-client", "https://kubernetes.default.svc/api",
+				client.FromKubernetesPod(namespace, "demo-client"),
+				client.WithCACert(caCert),
+				client.WithHeader("Authorization", fmt.Sprintf("Bearer %s", token)),
+			)
 			g.Expect(err).ToNot(HaveOccurred())
-			g.Expect(stdout).To(Equal("200"))
+			g.Expect(stdout).To(ContainSubstring(`"kind": "APIVersions"`))
```
Comment (michaelbeaumont) [inline test/e2e_env/kubernetes/k8s_api_bypass/k8s_api_bypass.go:93]: Maybe this deserves a comment? It's not clear to me why this means the test was successful

File: test/e2e_env/kubernetes/meshtimeout/meshtimeout.go [hunk 3]
```diff
@@ -48,21 +48,30 @@ func MeshTimeout() {
...
+				kubernetes.Cluster, "demo-client", fmt.Sprintf("test-server_%s_svc_80.mesh", namespace),
+				client.FromKubernetesPod(namespace, "demo-client"),
+				client.WithHeader("x-set-response-delay-ms", "5000"),
+				client.WithMaxTime(10), // we don't want 'curl' to return early
+			)
 			g.Expect(err).ToNot(HaveOccurred())
-			g.Expect(stdout).To(ContainSubstring("upstream request timeout"))
-		}).WithTimeout(30 * time.Second).Should(Succeed())
+			g.Expect(response.ResponseCode).To(Equal(504))
+		}, "30s", "1s").Should(Succeed())
```
Comment (michaelbeaumont) [inline test/e2e_env/kubernetes/meshtimeout/meshtimeout.go:74]: Didn't we set this as a default somewhere?

File: test/framework/client/collect.go [hunk 4]
```diff
@@ -238,11 +270,20 @@ func CollectResponse(
...
-			return types.EchoResponse{}, err
+			return "", "", err
 		}
 	}
 
-	stdout, stderr, err := cluster.Exec(opts.namespace, appPodName, container, cmd...)
+	return cluster.Exec(opts.namespace, appPodName, container, cmd...)
+}
+
+func CollectResponse(
```
Comment (michaelbeaumont) [inline test/framework/client/collect.go:280]: Not introduced by this PR but this should actually probably be called `CollectEchoResponse` and the other just `CollectResponse`

**Review by michaelbeaumont — APPROVED** [review]

**Review by jakubdyszkiewicz — APPROVED** [review]

File: test/e2e_env/kubernetes/meshtimeout/meshtimeout.go (hunk 3 above)
Comment (lobkovilya) [reply test/e2e_env/kubernetes/meshtimeout/meshtimeout.go:74]: Yes, we have, but I think explicit values are preferable. IIRC @jakubdyszkiewicz has an opinion on that

**Pull request merged**